src/cmd/cmd_c.ml

```diff
@@ -160,24 +159,30 @@ let metadata ~workspace arch property files : unit Result.t =
     Ok ()
   in
   let fpath = Fpath.(workspace / "test-suite" / "metadata.xml") in
-  let* (_exists : bool) = OS.Dir.create ~path:true (Fpath.parent fpath) in
   let* res = OS.File.with_oc fpath out_metadata { arch; property; files } in
   res
 
 let cmd ~debug ~arch ~property ~testcomp:_ ~workspace ~workers ~opt_lvl
   ~includes ~files ~profiling ~unsafe ~optimize ~no_stop_at_failure ~no_value
   ~no_assert_failure_expression_printing ~deterministic_result_order ~fail_mode
   ~concolic ~eacsl ~solver ~profile ~model_output_format
-  ~(entry_point : string option) ~invoke_with_symbols : unit Result.t =
+  ~(entry_point : string option) ~invoke_with_symbols ~out_file : unit Result.t
+    =
+  let workspace =
+    Option.value ~default:(Cmd_utils.tmp_dir "owi_c_%s") workspace
+  in
+  let* _ = OS.Dir.create Fpath.(workspace / "test-suite") in
```

Comment:
Can you add a type annotation so that we don't get a partial-application without noticing? (You could also name the result so it is easier to understand).